include/pisa/codec/block_codec.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <cstdint>
+#include <memory>
+#include <string_view>
+#include <vector>
+
+namespace pisa {
+
+/**
+ * Block codecs encode and decode a list of integers. This is in opposition to a streaming codec,
+ * which can encode and decode values one by one.
+ */
+class BlockCodec {
+  public:
+    /**
+     * Encodes a list of `n` unsigned integers and appends them to the output buffer.
+     */
+    virtual void encode(
+        std::uint32_t const* in, std::uint32_t sum_of_values, std::size_t n, std::vector<uint8_t>& out
```

Comment:
Why are we using the `std::uint` notation now? I have been seeing this more recently but I know it can break some builds with older compilers, may not matter for ours.

Reply:
These are part of the stdlib, and as such are in `std` namespace:

![image](https://github.com/user-attachments/assets/f5b413c0-eb60-4405-847b-5dbfdc93d98e)

Although some (most?) compilers have `typedef`s outside of `std` (I assume for C compatibility), it's not part of the standard.

One thing I've seen issues with is that these definitions (non-std ones) appear magically with certain imports, but not always the same with each compiler, so there's cases where it works for GCC and then crashes for Clang because Clang includes them in different headers.

Bottom line, I think it's best to stick with the standard and just add `std` namespaces to it.